### ISSUE TITLE
Modernizing the package, synchronizing with website

### ISSUE THREAD

**Issue by 5argon** [5argon/NotchSolution#43]
At the same time to make this an Asset Store package, I want to setup "the dream pipeline" such that :

- The structure follows https://docs.unity3d.com/Manual/cus-layout.html and https://www.youtube.com/watch?v=22HIEQTyozQ (for Samples folder)
- There is an `.md` documentation inside the package itself, so you know how to use everything even without internet and even inside IDE. (This is the present state)
- Design those `.md` so that it could generate a website/most part of website that I will self host. This website will provide much better SEO than this GitHub page. (And more colorful identity, etc.)
- The same but also generates Asset Store promotional text easily. Especially for images, there should be a pipeline that take some pictures from the `md` documentation to be Asset Store images.
- The generator could possibly run through the `Runtime` folder and generate API documentation as well as I have already put a lot of C# XML comments in. Not sure how.
- The website could be opened by clicking `View Documentation` in UPM window.
- Also `View changelog` should work (how?). The corresponding changelog in the website must be able to be generated from `CHANGELOG.md` that I am keeping currently.
- Use the `Release` section in GitHub to host all versions of Asset Store releases because Asset Store still don't have previous version function.

In summary, I want minimal effort maintaining the website as much as possible such that the package itself is a single source of truth.

**Comment by 06Games** [comment]
For the website, you can simply use [Wiki.js](https://wiki.js.org/) or another lighter alternative, such as [BookStack](https://www.bookstackapp.com/).
Regarding the documentation, it could be generated [thanks to Roslyn](https://stackoverflow.com/a/15768583/10792036).
I completely agree with you about using the `Release` section of Github which is a simple way to view the different versions of the project.

**Comment by 5argon** [comment]
I have decided to use [DocFx](https://dotnet.github.io/docfx/). It is quite nice. I am trying to learn it now + change code docs so it could gen the web properly. It seems to also supports "merging" `md` together with generated code documentation + standalone `.md` as its own page. Plus Unity official packages use it.

![image](https://user-images.githubusercontent.com/5653276/68675064-b0050980-0589-11ea-94ee-a0c23d5bdd54.png)

**Comment by 5argon** [comment]
Finally I have figured out how to structure the project in a way that the doc and code stay on their old folder and the website could be generated from it. (Even CHANGELOG.md can be crammed into the site on the build as a page, DocFx is pretty flexible.)

You don't have to pay attention to the `docfx` folder as some reference is an absolute path to the override template in my machine and I don't want to bake them with this repo. (it's unrelated, as it will be used for my other packages)

Plus I have override the DocFx default theme to improve typography.  It's now done : http://exceed7.com/notch-solution so I will close this issue!

![image](https://user-images.githubusercontent.com/5653276/68867094-deb8e680-0727-11ea-88ed-117211685c36.png)
